### ISSUE TITLE
(recitation 7) Method implementation of equals

### ISSUE THREAD

**Issue by wentinggy** [nus-cs2030/2021-s2#185]
Good Morning everyone! can i clarify that when the question asks us to implement equals method, it will always be implemented like this?
<img width="1437" alt="Screenshot 2021-04-07 at 5 52 13 PM" src="https://user-images.githubusercontent.com/77621160/113848572-36e40300-97cb-11eb-8e59-fdd52e199b19.png">
The purpose of line 42: Box<?> otherBox = (Box<?>) other; is to make it the most general type? 
thank you, have a nice day!

**Comment by ARPspoofing** [comment]
I would say the top part is a boilerplate code but the bottom part (where this.isPresent()) depends on the question. Prof henry mentioned something similar to this in the recent recitation where the bottom part will change if we are checking for Functors etc.

**Comment by kuro1511** [comment]
Most likely it will be similar to this implementation, but in the end it will depend on what the question ask for

**Comment by gableejh** [comment]
The format should be similar, i.e. checking if this is equal to the object, then instanceof otherwise false

**Comment by wentinggy** [comment]
ohh okay thanks!

**Comment by IntrovertHedgehog** [comment]
It can be simplified, but the general idea is the same

**Comment by mltnsia** [comment]
Yes this is generally the case but always depends on the question

**Comment by arnav0401** [comment]
This is generally how the **equals** method is implemented! It is good practice to begin by comparing this with 'other' and then specifically comparing the more specific types. i.e the start of this code is always the same, then the below part can be adjusted based on what the question demands
